eng/common/pipelines/templates/steps/sparse-checkout.yml

```diff
@@ -40,21 +59,33 @@ steps:
 
               Write-Host "git clone --no-checkout --filter=tree:0 https://github.com/$($repository.Name) ."
               git clone --no-checkout --filter=tree:0 https://github.com/$($repository.Name) .
+              if ($LASTEXITCODE) {
+                return (Clone $repository && Pop-Location)
```

Comment:
I anticipate objections to my stubborn desire for brevity here.

Reply:
Lets avoid the Pop-Location everywhere and just add a finally block in this function. Also why the return statement the Clone function doesn't return anything. Should we just call Clone and then an empty return?

Reply:
I can fix the pop-location. I only had the return on the same line to save an extra line, but with two lines instead of three it's not as bad.